vtex-apps/order-manager
# Fix stale orderForm in localStorage when id differ

#### What problem is this solving?
This PR basically re-enables what we removed in #33. Since #37 we added a `heuristics.ts` file that should contain all heuristics we use to replace the local orderForm with the incoming remote one, so I moved the condition there.

Also, I added a check for `orderFormOptimizationEnabled` because this will only work on stores that have this enabled, else we would be going on the same route that #31 took. This flag tells us that only one of the `OrderFormProvider`s will be used in the store, so we won't have the concurrency problem the original PR had.

#### How should this be manually tested?
[Workspace](https://chkio--checkoutio.myvtex.com/)

Steps:

1. Add something to your cart.
2. Delete your `checkout.vtex.com` cookie and reload.
3. Check if your minicart is empty after the query resolves.

Another test I usually do to ensure I don't break offline support is:

1. Enter the store.
2. Turn offline mode on in your browser ("Work Offline" in Firefox, network type "Offline" in Chrome's network tab).
3. Add an item to your cart.
4. Close the tab, turn offline mode off.
5. Open the store again, and wait for the queries to resolve.

You should end up with the item you added while offline in your cart, and **should not** see the items count in the minicart badge change during the second load.

#### Checklist/Reminders


#### Type of changes


✔️ | Type of Change
---|---
✔️ | Bug fix 
_ | New feature 
_ | Breaking change 
_ | Technical improvements
